### PR TITLE
Disable tests on Windows [HZ-2407]

### PR DESCRIPTION
**Pull request by orcunc**
Disable tests that use testcontainers on Windows

Fixes :  https://github.com/hazelcast/hazelcast/issues/24414
Jira : https://hazelcast.atlassian.net/browse/HZ-2407

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by olukas — CHANGES_REQUESTED** [review]

File: hazelcast/src/test/java/com/hazelcast/dataconnection/impl/ListResourcesTest.java [hunk 1]
```diff
@@ -79,6 +82,16 @@ public static Object[] params() {
     @Parameter(1)
     public DataConnectionResource[] expectedResources;
 
+    @Before
+    public void setUp() {
+        if (OsHelper.isWindows()) {
```
Comment (olukas) [inline hazelcast/src/test/java/com/hazelcast/dataconnection/impl/ListResourcesTest.java:87]: Why don't we rather use `DockerTestUtil.dockerEnabled()` method?

**Review by olukas — CHANGES_REQUESTED** [review]

File: hazelcast/src/test/java/com/hazelcast/dataconnection/impl/ListResourcesTest.java (hunk 1 above)
Comment (orcunc) [reply hazelcast/src/test/java/com/hazelcast/dataconnection/impl/ListResourcesTest.java:87]: You mean something like this ? 
```
@Before
public void setUp() {
  // If Windows
  if (!DockerTestUtil.dockerEnabled()) {
    assumeThat(provider)
                    .isInstanceOfAny(
                            H2DatabaseProvider.class
      );
   }
 }
````

Comment (orcunc) [reply hazelcast/src/test/java/com/hazelcast/dataconnection/impl/ListResourcesTest.java]: Here is a more generic approach
rev : f52cd34ac3ca97f420580adc96a861933039a44b

File: hazelcast/src/test/java/com/hazelcast/test/DockerTestUtil.java [hunk 2]
```diff
@@ -25,4 +29,15 @@ public static boolean dockerEnabled() {
     public static void assumeDockerEnabled() {
         assumeTrue(DockerTestUtil.dockerEnabled());
     }
+
+    public static void assumeProviderCanLaunch(TestDatabaseProvider provider) {
```
Comment (olukas) [inline hazelcast/src/test/java/com/hazelcast/test/DockerTestUtil.java:33]: `assumeProviderCanLaunch` -> `assumeTestDatabaseProviderCanLaunch`

**Review by olukas — APPROVED** [review]
LGTM, just rename the method name please as mentioned in the comment.

**Comment by ldziedziul** [comment]
@orcunc I think that `DockerTestUtil` is wrong place for `assumeTestDatabaseProviderCanLaunch` and `assumeTestDatabaseProviderIsNotInstanceOf`. Could you move them somewhere else?

**Pull request merged**